gobblin-runtime/src/main/java/gobblin/runtime/Task.java

```diff
@@ -279,7 +280,8 @@ private void publishTaskData() throws IOException {
       @SuppressWarnings("unchecked")
       Class<? extends DataPublisher> dataPublisherClass = (Class<? extends DataPublisher>) Class.forName(
           this.taskState.getProp(ConfigurationKeys.DATA_PUBLISHER_TYPE, ConfigurationKeys.DEFAULT_DATA_PUBLISHER_TYPE));
-      DataPublisher publisher = closer.register(DataPublisher.getInstance(dataPublisherClass, this.taskState));
+      SingleTaskDataPublisher publisher =
```

Comment:
This requires data publishers to be of type SingleTaskDataPublisher. If any user is already using a DataPublisher that is not of this type, Gobblin would fail.